src/platform/silabs/rs911x/wfx_sl_ble_init.cpp

```diff
@@ -20,51 +20,41 @@
  *    See the License for the specific language governing permissions and
  *    limitations under the License.
  */
+#include <lib/support/logging/CHIPLogging.h>
+#include <platform/CHIPDeviceLayer.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
 #include "wfx_sl_ble_init.h"
-#include "ble_config.h"
-#include "cmsis_os2.h"
-#include "silabs_utils.h"
+#ifdef __cplusplus
+}
+#endif
+
 // Global Variables
 rsi_ble_t att_list;
-sl_wfx_msg_t event_msg;
+BleEvent_t bleEvent;
+static osMessageQueueId_t sBleEventQueue = NULL;
 
-extern osSemaphoreId_t sl_ble_event_sem;
-
-// Memory to initialize driver
-uint8_t bt_global_buf[BT_GLOBAL_BUFF_LEN];
-const uint8_t ShortUUID_CHIPoBLEService[] = { 0xF6, 0xFF };
+void InitBleEventQueue()

```

Comment:
Your queue should be implemented the other way around. Here you create the queue and post to the queue in in this file  `(wfx_sl_ble_init.cpp)`, but then you have BleManagerImpl.cpp that Gets the queue handle from a public function to then process/dequeue it. Nothing here would prevent another instance/process from also calling `GetBleEventQueue` and trying to process the events.


BleManagerImpl should create, own and process the queue and provide an API for other processis to post on the queue. The queue itself should not be accessible by others.

Reply:
Updated the PR